packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-base-chart.tsx

```diff
@@ -540,15 +540,18 @@ export class ScWebglBaseChart {
    * Provide no `hasDataChanged` to prevent a re-processing of the chart scenes.
    */
   onUpdate = (
-    { start, end }: { start: Date; end: Date },
-
+    {
+      start,
+      end,
+      shouldBlockDateRangeChangedEvent,
+    }: { start: Date; end: Date; shouldBlockDateRangeChangedEvent?: boolean },
     hasDataChanged: boolean = false,
     hasSizeChanged: boolean = false,
     hasAnnotationChanged: boolean = false,
     shouldRerender: boolean = false
   ) => {
     const hasViewPortChanged = this.start.getTime() !== start.getTime() || this.end.getTime() !== end.getTime();
-    if (hasViewPortChanged) {
+    if (hasViewPortChanged && !shouldBlockDateRangeChangedEvent) {
```

Comment:
Does this implementation work with Gestures (zoom in, zoom out, pan) for charts that start in live-mode? We want to make sure we aren't blocking change events for gestures for charts that start in live mode.

Reply:
And by gestures, more specifically, i mean widget controls - https://synchrocharts.com/#/Introduction.

Reply:
Yep! This works with gestures! We want to emit the `dateRangeChanged` event any time the viewport changes, unless we explicitly block it. When gestures happen we aren't blocking the event.